### ISSUE TITLE
Dynamic configuration for retryConfig

### ISSUE THREAD

**Issue by ranarula** [resilience4j/resilience4j#1043]
Resilience4j version: 1.5.0

Java version: 11

Thanks for raising a Resilience4j issue.
Is there a way to dynamically update the interval function for the retry module. 

Use case : I want to perform retries based on the RETRY_AFTER for my http requests. Currently the only way to configure retry is by pre-configuring an IntervalFunction on the retryConfig. 

I have configured the `resultPredicate` to look for the header to retry but I would want to retry based on the time specified in the header field. 
I was wondering if there is a way to configure the RetryConfig for this scenario.

**Comment by RobWin** [comment]
Hi,
that means you want to reconfigure the `resultPredicate` at runtime?

**Comment by ranarula** [comment]
not exactly - 

I use the result predicate to verify if we need to dynamically delay (by checking the value of the response header `Retry-After`) The delay value to be used is from the value sent in the header. 

Please check section 14.37 for https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html for more details.

My query was if there is a way to override the IntervalFunction to use this value and retry after the time as received in the `Retry-After` header.

**Comment by RobWin** [comment]
No, but it would a nice new feature. Even if the IntervalFuntion would be mutable, it wouldn't work, because the function is used for all requests which are retried. As far as I understand you want to have a different intervalfunction per request/response header, correct? And the `Retry-After` header can be different for every response?

**Comment by RobWin** [comment]
For example:

`(Integer attempt, T result) -> Long waitDuration`

**Comment by ranarula** [comment]
Right exactly, a BiFunction support for IntervalFunction will do the trick

**Comment by cosminseceleanu** [comment]
Hi @RobWin, any news on this? Is this feature on the roadmap?

**Comment by cosminseceleanu** [comment]
@RobWin I want to open a pull request but before starting to write some code I have question. Should I use a single function as you mentioned or two functions 'intervalOnResultFunction' and `intervalOnExceptionPredicate`?
I'm asking because for predicates the current API uses two.

**Comment by RobWin** [comment]
Sry for the late response. 
I think I would even prefer two functions:
`intervalFunctionOnResult` and `intervalFunctionOnException`.

If a result must be retried, either `intervalFunctionOnResult` or as a fallback `interfalFunction` is used. 
If an exception must be retried, either `intervalFunctionOnException` or as a fallback `interfalFunction` is used. 

What do you think?

**Comment by cosminseceleanu** [comment]
Sounds good to me, I also prefer a single function.

**Comment by RobWin** [comment]
Looking forward to your  PR ;)
